File: app/src/main/java/com/acdos/comp41690/ui/solar/SolarViewModel.java
package com.acdos.comp41690.ui.solar;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class SolarViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public SolarViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is slideshow fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/com/example/help/ui/solar/SolarViewModel.kt
package com.example.help.ui.solar

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class SolarViewModel : ViewModel() {

    private val _text = MutableLiveData<String>().apply {
        value = "This is solar Fragment"
    }
    val text: LiveData<String> = _text
}